tests/CMakeLists.txt

```diff
@@ -0,0 +1,163 @@
+    PREFIX ${CMAKE_CURRENT_BINARY_DIR}/gtest
+    CMAKE_ARGS -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+    # Disable install step
+    INSTALL_COMMAND "")
+  set (gtest_root "${CMAKE_CURRENT_BINARY_DIR}/gtest/src/gtest/googletest")
+  message(STATUS "gtest locatet at: ${gtest_root}")
+
+  # Create a libgtest target to be used as a dependency by test programs
+  add_library(libgtest IMPORTED STATIC GLOBAL)
+  add_dependencies(libgtest gtest)
+
+  # Get GTest source and binary directories from CMake project
+  ExternalProject_Get_Property(gtest source_dir binary_dir)
+
+  # Set libgtest properties
+  set_target_properties(libgtest PROPERTIES
+    "IMPORTED_LOCATION" "${binary_dir}/googlemock/gtest/libgtest.a"
+    "IMPORTED_LINK_INTERFACE_LIBRARIES" "${CMAKE_THREAD_LIBS_INIT}")
+
+  # Export gtest variables
+  set(GTEST_ROOT ${gtest_root} PARENT_SCOPE)
+  set(GTEST_INCLUDE_DIRS ${gtest_root}/include)
+  set(GTEST_MAIN_LIBRARY gtest_main PARENT_SCOPE)
+  set(GTEST_BOTH_LIBRARIES libgtest)
+  set(GTEST_FOUND true)
+  message(STATUS "gtest include dir: ${GTEST_INCLUDE_DIRS}")
+endif()
+
+# Include directories
+include_directories(
+  ${GTEST_INCLUDE_DIRS}
+  ${OPAE_INCLUDE_DIR}
+  ${OPAE_SDK_SOURCE}/libopae/src
+  ${OPAE_SDK_SOURCE}/tools/libopae++
+  ${OPAE_SDK_SOURCE}/tools/c++utils
+  ${Boost_INCLUDE_DIRS}
+  ${OPAE_SDK_SOURCE}/common/include
+  ${OPAE_SDK_SOURCE}/tests)
+
+add_library(commonlib SHARED common_test.h common_test.cpp)
+target_link_libraries(commonlib opae-c ${GTEST_BOTH_LIBRARIES} ${jsoncpp_LIBRARIES})
+target_include_directories(commonlib PUBLIC
+  $<BUILD_INTERFACE:${GTEST_INCLUDE_DIRS}>
+  $<BUILD_INTERFACE:${OPAE_INCLUDE_DIR}>
+  $<BUILD_INTERFACE:${jsoncpp_INCLUDE_DIRS}>
+  $<INSTALL_INTERFACE:include>
+  PRIVATE $<BUILD_INTERFACE:${OPAE_SDK_SOURCE}/libopae/src>)
+
+# Test binary
+set(SRC gtmain.cpp
+  jsonParser.cpp
+  unit/gtOpenClose_base.cpp)
+add_executable(gtapi ${SRC})
+target_include_directories(gtapi PUBLIC
+  $<BUILD_INTERFACE:${GTEST_INCLUDE_DIRS}>
+  $<BUILD_INTERFACE:${OPAE_INCLUDE_DIR}>
+  $<INSTALL_INTERFACE:include>
+  PRIVATE $<BUILD_INTERFACE:${OPAE_SDK_SOURCE}/libopae/src>)
+target_link_libraries(gtapi commonlib safestr opae-c json-c uuid ${GTEST_BOTH_LIBRARIES}
+  opae-c++-utils opae-c++)
+configure_file(${PROJECT_SOURCE_DIR}/configuration.json
+  ${CMAKE_BINARY_DIR}/tests/configuration.json
+  COPYONLY)
+
+if(THREADS_HAVE_PTHREAD_ARG)
+  target_compile_options(PUBLIC commonlib "-pthread")
+  target_compile_options(PUBLIC gtapi "-pthread")
+endif()
+
+if(CMAKE_THREAD_LIBS_INIT)
+  target_link_libraries(gtapi "${CMAKE_THREAD_LIBS_INIT}")
+endif()
+
+# build mock driver fake directory structure
+add_custom_target(mock-sysfs-prepare)
+add_custom_command(TARGET mock-sysfs-prepare POST_BUILD
+  COMMAND cmake -E copy ${CMAKE_CURRENT_SOURCE_DIR}/mock_sys_tmp-1socket-nlb0.tar.gz /tmp
+  COMMAND tar xzvf /tmp/mock_sys_tmp-1socket-nlb0.tar.gz -C /tmp --strip 1)
+
+# build mock driver
+add_library(mock SHARED mock.c)
```

Comment:
mock is pretty generic... would it make sense to rename this library as `opae-mock` or `opae-mock-sysfs`?